scripts/launch_workflow_cf/README.md

```diff
@@ -1,28 +1,122 @@
+
+**New to Terra groups and permissions?** Don't start here. Instead see:
+
+* [Pet accounts and proxy groups – Terra Support](https://support.terra.bio/hc/en-us/articles/360031023592-Understanding-and-setting-up-a-proxy-group)
+* [Accessing advanced GCP features in Terra – Terra Support](https://support.terra.bio/hc/en-us/articles/360051229072)
+
+# Setup
+
+## Configure the Terra Workflow
+
+1. Create or identify the Terra workflow you want to trigger via a Cloud Function.
+    * For a simple example, see [example.wdl](./example.wdl).
+1. Upload an entity set of all but one of the parameters for your workflow. The one parameter absent from the entity set should be the one that is the path to the Cloud Storage file that will trigger the workflow.
+    * Create this entity set even if the set would only contain one entity (one row of parameter values).
+    * For a simple example, see [example.tsv](./example.tsv).
+1. **IMPORTANT** Make sure you have used the Terra User Interface to run your workflow successfully at least once using the entity set.
+
+## Configure the Cloud and Terra execution environments
+
+1. Create or identify the Google Cloud Platform project where this Cloud Function will live / be billed.
+    1. Enable the [Cloud Functions API](https://console.developers.google.com/apis/library/cloudfunctions.googleapis.com) for that project.
+    1. Enable the [Secret Manager API](https://console.cloud.google.com/marketplace/product/google/secretmanager.googleapis.com) for that project.
+    1. Enable the [Cloud Build API](https://console.developers.google.com/apis/library/cloudbuild.googleapis.com) for this project.
+1. Create or identify the bucket in that project that will be used for files to trigger the Cloud Function.
+1. Create a new [service account](https://console.cloud.google.com/iam-admin/serviceaccounts) within that project. It will look like `<your service account name>@<your project id>.iam.gserviceaccount.com`.
+1. Download a JSON key for the newly created service account and store it in a safe place.
+1. Upload the JSON key to [Secret Manager](https://console.cloud.google.com/security/secret-manager).
+1. For the newly created secret, in column "Actions", choose "Copy Resource ID". You'll need this value for the deployment step and it will look like: `projects/<your project number>/secrets/<your secret name>/versions/1`.
+1. Give the service account the needed permissions to items within the project.
+    1. Grant role `Secret Manager Secret Accessor` to the service account.
+    1. Grant role `Storage Object Viewer` to the service account on the bucket that will trigger the Cloud Function.
+1. [Register the service account](https://github.com/broadinstitute/terra-tools/tree/master/scripts/register_service_account) in Terra.
+1. [Create a new Terra group](https://app.terra.bio/#groups) to hold only the service account. It will look like `
```

Comment:
genius to create a Terra group for this rather than having to find the proxy group. this readme is *incredible*, Nicole, wow.